### PR TITLE
[exporter/Datadogexporter] Use Agent modules for validating hostname

### PR DESCRIPTION
**Pull request by dineshg13**
**Description:** <Describe what has changed.>
Use Agent modules for validating hostname

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by mx-psi — COMMENTED** [review]
Code looks good, can approve once CI passes

File: exporter/datadogexporter/internal/hostmetadata/internal/system/package_test.go
Comment (songy23) [inline exporter/datadogexporter/internal/hostmetadata/internal/system/package_test.go:14]: What is this? Can you add a tracking issue?

Comment (dineshg13) [reply exporter/datadogexporter/internal/hostmetadata/internal/system/package_test.go:14]: Agent module is logging error here https://github.com/DataDog/datadog-agent/blob/78ca7aad3407f9202fc0ae3f81623907a9e32691/pkg/util/hostname/validate/validate.go#L40
I don't see a need to track it.

Comment (mx-psi) [reply exporter/datadogexporter/internal/hostmetadata/internal/system/package_test.go:14]: @crobert-1 Could you help us with the best way to deal with this? (e.g. is there a template for tracking issues?)

Comment (mx-psi) [reply exporter/datadogexporter/internal/hostmetadata/internal/system/package_test.go:14]: > I don't see a need to track it.

We aim to have no leaks at all in the Collector, this should be tracked and eventually fixed in the Agent.

Comment (dineshg13) [reply exporter/datadogexporter/internal/hostmetadata/internal/system/package_test.go:14]: Created internal jira to track this work in Agent.

Comment (crobert-1) [reply exporter/datadogexporter/internal/hostmetadata/internal/system/package_test.go:14]: Please add frequency and information to this issue: https://github.com/cihub/seelog/issues/182

Also, please add a reference in this file to that issue so that we have future context for why this ignore is necessary.

**Review by songy23 — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
@crobert-1 will wait until Monday in case you want to review

**Review by crobert-1 — APPROVED** [review]

**Pull request merged**
